### PR TITLE
Fix NuGet/Home#776: MultipartWebRequest::CreateMultipartRequest should us "\r\n" explicitly and not Environment.NewLine

### PR DESCRIPTION
**Pull request by feiling**

### CODE REVIEW

File: src/NuGet.CommandLine/MultipartWebRequest.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            using (Stream stream = request.GetRequestStream())
+            {
+                foreach (var item in _formData)
+                {
+                    string header = String.Format(CultureInfo.InvariantCulture, FormDataTemplate, boundary, item.Key, item.Value);
+                    byte[] headerBytes = Encoding.UTF8.GetBytes(header);
+                    stream.Write(headerBytes, 0, headerBytes.Length);
+                }
+
+                byte[] newlineBytes = Encoding.UTF8.GetBytes("\r\n");
```
Comment (feiling) [inline src/NuGet.CommandLine/MultipartWebRequest.cs]: This is the fix.

Comment (feiling) [inline src/NuGet.CommandLine/MultipartWebRequest.cs:63]: The fix is to use "\r\n" here.

File: src/NuGet.CommandLine/MultipartWebRequest.cs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+            foreach (var item in _formData)
+            {
+                string header = String.Format(CultureInfo.InvariantCulture, FormDataTemplate, boundary, item.Key, item.Value);
+                byte[] headerBytes = Encoding.UTF8.GetBytes(header);
+
+                totalContentLength += headerBytes.Length;
+            }
+
+            byte[] newlineBytes = Encoding.UTF8.GetBytes("\r\n");
```
Comment (feiling) [inline src/NuGet.CommandLine/MultipartWebRequest.cs:94]: Also, we need to use "\r\n" here.

**Comment by feiling** [comment]
All other changes are just copies from NuGet.V2.

@deepakaravindr @emgarten @zhili1208 @yishaigalatzer

File: src/NuGet.CommandLine/MultipartWebRequest.cs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+namespace NuGet.CommandLine
+{
+    /// <remarks>
+    /// Based on the blog post by Travis Illig at http://www.paraesthesia.com/archive/2009/12/16/posting-multipartform-data-using-.net-webrequest.aspx
+    /// </remarks>
+    public class MultipartWebRequest
+    {
+        private const string FormDataTemplate = "--{0}\r\nContent-Disposition: form-data; name=\"{1}\"\r\n\r\n{2}\r\n";
+        private const string FileTemplate = "--{0}\r\nContent-Disposition: form-data; name=\"{1}\"; filename=\"{2}\"\r\nContent-Type: {3}\r\n\r\n";
```
Comment (deepakaravindr) [inline src/NuGet.CommandLine/MultipartWebRequest.cs:16]: Break long lines here and in other places in the CR ( < 120 chars per line)

File: src/NuGet.CommandLine/MultipartWebRequest.cs [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+                FileFactory = fileFactory,
+                FieldName = fieldName,
+                ContentType = contentType,
+                ContentLength = length
+            });
+        }
+
+        public void CreateMultipartRequest(WebRequest request)
+        {
+            string boundary = "---------------------------" + DateTime.Now.Ticks.ToString("x", CultureInfo.InvariantCulture);
```
Comment (deepakaravindr) [inline src/NuGet.CommandLine/MultipartWebRequest.cs:50]: Use var here and in other places in the CR as applicable. Another example is using (Stream stream below)

File: src/NuGet.CommandLine/MultipartWebRequest.cs [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            }
+        }
+
+        private long CalculateContentLength(string boundary)
+        {
+            long totalContentLength = 0;
+
+            foreach (var item in _formData)
+            {
+                string header = String.Format(CultureInfo.InvariantCulture, FormDataTemplate, boundary, item.Key, item.Value);
```
Comment (deepakaravindr) [inline src/NuGet.CommandLine/MultipartWebRequest.cs:88]: Use smaller case string for the static method string.Format. Please do the same for all other uses of static methods of string.

File: src/NuGet.CommandLine/MultipartWebRequest.cs [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        private readonly List<PostFileData> _files;
+
+        public MultipartWebRequest()
+            : this(new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase))
+        {
+        }
+
+        public MultipartWebRequest(Dictionary<string, string> formData)
+        {
+            _formData = formData;
```
Comment (emgarten) [inline src/NuGet.CommandLine/MultipartWebRequest.cs:28]: Check for null on formData.

File: src/NuGet.CommandLine/MultipartWebRequest.cs [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+            _files = new List<PostFileData>();
+        }
+
+        public void AddFormData(string key, string value)
+        {
+            _formData.Add(key, value);
+        }
+
+        public void AddFile(Func<Stream> fileFactory, string fieldName, long length, string contentType = "application/octet-stream")
```
Comment (emgarten) [inline src/NuGet.CommandLine/MultipartWebRequest.cs:37]: don't use optional parameters

**Pull request closed without merging**